### ISSUE TITLE
system access violation exception in cvtrack.cpp

### ISSUE THREAD

**Issue by ispysoftware** [emgucv/cvblob#1]
\emgucv34\Emgu.CV.Extern\cvblob\libcvblob\cvBlob\cvtrack.cpp

- throwing system access violation exception at line 292 because track is null.
track is not assigned by for loop @ line 260 because a = 0
(maxy = miny or maxx=minx)

I think there might be a bug in the area calculation - if x and y are zero based then need to add 1 to each

eg: (line 264)

unsigned int a = (t->maxx - t->minx + 1)*(t->maxy - t->miny + 1);

**Comment by emgucv** [comment]
Does it means there will be an access violating for updateTracks when all blobs has either maxy == miny or maxx==minx?

I feel like it is better to put a safe guard around 292 and 300:
```
	  if (track)
	  {
		  track->label = blob->label;
		  track->centroid = blob->centroid;
		  track->minx = blob->minx;
		  track->miny = blob->miny;
		  track->maxx = blob->maxx;
		  track->maxy = blob->maxy;
		  if (track->inactive)
			  track->active = 0;
		  track->inactive = 0;
	  }
```
So we don't update the track if all blobs has 0 area (by which I means maxy == miny or maxx==minx)

The code in line 302-315 will mark all tracks inactive afterwards, which make sense.

**Comment by ispysoftware** [comment]
yeah that could also work - i'm not sure if it should be deleted or not but
the area calculated is wrong as an area with maxy=miny (ie 1 px high)
should have >0 area. It needs to add 1 to both dimensions which means the
area should never be 0.

This is what i've done by the way and it seems to be working fine

**Comment by emgucv** [comment]
I have seem `area = (maxx - minx) * (maxy - miny)` to be a more common way to calculate the area. 
This is done such that an empty blob / rectangle will have an area of 0.
An empty blob will usually have maxx = minx = maxy = miny = 0, using the above equation they will have area of 0. using the equation 
`int a = (t->maxx - t->minx + 1)*(t->maxy - t->miny + 1);`
it will have area of 1, which is un-expected.

I will try to use the safe guard implementation I have proposed above, this way for existing blobs that do not cause access violating, the area computation remain the same. 

Do you have some test code that can cause the access violation, such that I can use for testing?

It is still strange to see why there are blobs detected that has either minx==maxx or miny==maxy in the first place.

**Comment by ispysoftware** [comment]
i don't have test code as it just seems to be a random thing where it detects an area 1px in height or width.

if it was using this method of setting maxx = minx = maxy = miny = 0 to signify an empty blob then it would have been crashing with the original bug constantly.
